django/thunderstore/security/utils.py

```diff
@@ -0,0 +1,140 @@
+import base64
+import binascii
+import json
+
+import requests
+from cryptography.hazmat.backends.openssl.backend import backend
+from cryptography.hazmat.primitives.asymmetric import ec
+from six import b, text_type
+
+
+class DERStringDecoder(object):
+    def __init__(self, initial_bytes):
+        self._buffer = initial_bytes
+        self._index = 0
+
+    def remaining_data(self):
+        return len(self._buffer) - self._index
+
+    def read(self, length):
+        new_index = self._index + length
+        if new_index > len(self._buffer):
+            raise ValueError(
+                "Index is over the buffer index: %d buffer length: %d"
+                % (new_index, len(self._buffer))
+            )
+        result = self._buffer[self._index : new_index]
+        self._index = new_index
+        return int(result, 16)
+
+    def decode(self):
+        comps = [str(x) for x in divmod(self.read(2), 40)]
+        v = 0
+        while self.remaining_data():
+            c = self.read(2)
+            v = v * 128 + (c & 0x7F)
+            if not (c & 0x80):
+                comps.append(str(v))
+                v = 0
+        decoded = ".".join(comps)
+        return decoded
+
+
+def unpem(pem):
+    if isinstance(pem, text_type):  # pragma: no branch
+        pem = pem.encode()
+
+    d = b("").join(
+        [l.strip() for l in pem.split(b("\n")) if l and not l.startswith(b("-----"))]
+    )
+    return base64.b64decode(d)
+
+
+def ec_points_from_der(key):
+    # Let's unDER
+    bits = binascii.hexlify(unpem(key))
+    # Sequence 1
+    assert bits[0:2] == b"30"
```

Comment:
Don't use assertions for runtime validation outside of tests, use if checks and raise exceptions instead. Ideally a custom exception in a case like this. For one thing, assertions are a bad idea for checks that should always be made as they can be disabled with a python CLI flag, but if you need to write error handling code, you don't want to be catching a generic `AssertionError`.

Reply:
Yeah, using the assertions felt kinda weird, but went for it on the draft, mainly because I wanted to get the whole thing working first. I'll make some changes.